File: src/main/java/edu/recepespire/service/InventoryService.java
package edu.recepespire.service;

import edu.recepespire.entity.InventoryItem;

import java.util.List;

public interface InventoryService {
    List<InventoryItem> getAllInventoryItems();
    List<InventoryItem> getLowStockItems();
    InventoryItem addToInventory(InventoryItem item);
    InventoryItem updateInventoryItem(Long id, InventoryItem item);
    void removeFromInventory(Long id);
}

File: recipe-spire/src/app/core/services/inventory.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { InventoryItem } from '../../shared/models/inventory.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class InventoryService {
  private apiUrl = `${environment.apiBaseUrl}/inventory`;

  constructor(private http: HttpClient) {}

  getAllInventoryItems(): Observable<InventoryItem[]> {
    return this.http.get<InventoryItem[]>(this.apiUrl);
  }

  getLowStockItems(): Observable<InventoryItem[]> {
    return this.http.get<InventoryItem[]>(`${this.apiUrl}/low-stock`);
  }

  addToInventory(item: InventoryItem): Observable<InventoryItem> {
    return this.http.post<InventoryItem>(this.apiUrl, item);
  }

  updateInventoryItem(id: number, item: InventoryItem): Observable<InventoryItem> {
    return this.http.put<InventoryItem>(`${this.apiUrl}/${id}`, item);
  }

  removeFromInventory(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}